data/transactions/verify/txn.go

```diff
@@ -214,13 +214,10 @@ func LogicSigSanityCheck(txn *transactions.SignedTxn, groupIndex int, groupCtx *
 		GroupIndex:     groupIndex,
 		MinTealVersion: &groupCtx.minTealVersion,
 	}
-	cost, err := logic.Check(lsig.Logic, ep)
+	err := logic.Check(lsig.Logic, ep)
 	if err != nil {
 		return err
 	}
-	if cost > int(groupCtx.consensusParams.LogicSigMaxCost) {
-		return fmt.Errorf("LogicSig.Logic too slow, %d > %d", cost, groupCtx.consensusParams.LogicSigMaxCost)
-	}
```

Comment:
this looks like can break backward compatibility. I think for TEAL v1-v3 we still need this check.

Reply:
I see now, thanks for pointing out